src/Umbraco.Web.UI.Client/src/views/languages/edit.controller.js

```diff
@@ -20,150 +19,118 @@
             $timeout(function () {
-                navigationService.syncTree({ tree: "languages", path: "-1" });
+                navigationService.syncTree({ tree: $routeParams.tree, path: [-1], activate: true });
             });
         }
 
         function save() {
+            // Set new language name to culture name
+            if (vm.isNew && vm.language.culture && vm.availableCultures) {
+                vm.language.name = vm.availableCultures[vm.language.culture];
+            }
 
             if (formHelper.submitForm({ scope: $scope })) {
-                vm.page.saveButtonState = "busy";
+                vm.saveButtonState = "busy";
 
-                //check if the culture is being changed
-                if (currCulture && vm.language.culture !== currCulture) {
-
-                    const changeCultureAlert = {
+                // Check if the culture is being changed
+                if (initCulture && vm.language.culture !== initCulture) {
+                    overlayService.open({
                         title: vm.labels.areYouSure,
                         view: "views/languages/overlays/change.html",
                         submitButtonLabelKey: "general_continue",
-                        submit: function (model) {
+                        submit: function () {
                             saveLanguage();
                             overlayService.close();
                         },
                         close: function () {
                             overlayService.close();
-                            vm.page.saveButtonState = "init";
+                            vm.saveButtonState = "init";
                         }
-                    };
-
-                    overlayService.open(changeCultureAlert);
-                }
-                else {
+                    });
+                } else {
                     saveLanguage();
                 }
             }
         }
 
         function saveLanguage() {
             languageResource.save(vm.language).then(function (lang) {
-
                 formHelper.resetForm({ scope: $scope });
 
                 vm.language = lang;
-                vm.page.saveButtonState = "success";
+                vm.saveButtonState = "success";
+
                 localizationService.localize("speechBubbles_languageSaved").then(function (value) {
                     notificationsService.success(value);
                 });
 
-                // emit event when language is created or updated/saved
-                var args = { language: lang, isNew: $routeParams.create ? true : false };
-                eventsService.emit("editors.languages.languageSaved", args);
-
-                back();
+                // Emit event when language is created or updated/saved
+                eventsService.emit("editors.languages.languageSaved", {
+                    language: lang,
+                    isNew: vm.isNew
+                });
 
+                vm.isNew = false;
```

Comment:
This now emits the event with a correct `isNew` flag and doesn't redirect the user back to the overview (this was probably done as work-around for this issue).